utils/setup_functions.php

```diff
@@ -0,0 +1,931 @@
+
+    $sSQL = 'select count(*) from (select getorcreate_country(make_standard_name(v), country_code) from (select country_code, skeys(name) as k, svals(name) as v from country_name) x where k ';
+    if (CONST_Languages) {
+        $sSQL .= 'in ';
+        $sDelim = '(';
+        foreach (explode(',', CONST_Languages) as $sLang) {
+            $sSQL .= $sDelim."'name:$sLang'";
+            $sDelim = ',';
+        }
+        $sSQL .= ')';
+    } else {
+        // all include all simple name tags
+        $sSQL .= "like 'name:%'";
+    }
+    $sSQL .= ') v';
+    pgsqlRunScript($sSQL);
+}
+
+function drop($aCMDResult)
+{
+    info('Drop tables only required for updates');
+    // The implementation is potentially a bit dangerous because it uses
+    // a positive selection of tables to keep, and deletes everything else.
+    // Including any tables that the unsuspecting user might have manually
+    // created. USE AT YOUR OWN PERIL.
+
+
+    // tables we want to keep. everything else goes.
+    $aKeepTables = array(
+                    '*columns',
+                    'import_polygon_*',
+                    'import_status',
+                    'place_addressline',
+                    'location_postcode',
+                    'location_property*',
+                    'placex',
+                    'search_name',
+                    'seq_*',
+                    'word',
+                    'query_log',
+                    'new_query_log',
+                    'spatial_ref_sys',
+                    'country_name',
+                    'place_classtype_*'
+                   );
+
+    $oDB =& getDB();
+    $aDropTables = array();
+    $aHaveTables = chksql($oDB->getCol("SELECT tablename FROM pg_tables WHERE schemaname='public'"));
+
+    foreach ($aHaveTables as $sTable) {
+        $bFound = false;
+        foreach ($aKeepTables as $sKeep) {
+            if (fnmatch($sKeep, $sTable)) {
+                $bFound = true;
+                break;
+            }
+        }
+        if (!$bFound) array_push($aDropTables, $sTable);
+    }
+
+    foreach ($aDropTables as $sDrop) {
+        if ($aCMDResult['verbose']) echo "dropping table $sDrop\n";
+        @pg_query($oDB->connection, "DROP TABLE $sDrop CASCADE");
+        // ignore warnings/errors as they might be caused by a table having
+        // been deleted already by CASCADE
+    }
+
+    if (!is_null(CONST_Osm2pgsql_Flatnode_File) && CONST_Osm2pgsql_Flatnode_File) {
+        if ($aCMDResult['verbose']) echo 'deleting '.CONST_Osm2pgsql_Flatnode_File."\n";
+        unlink(CONST_Osm2pgsql_Flatnode_File);
+    }
+}
+
+function didsomething($bDidSomething)
+{
+    if (!$bDidSomething) {
+        showUsage($aCMDOptions, true);
+    } else {
+        echo "Summary of warnings:\n\n";
+        repeatWarnings();
+        echo "\n";
+        info('Setup finished.');
+    }
+}
+
+// *********************************
+
+function pgsqlRunScriptFile($sFilename)
+{
+    global $aCMDResult;
```

Comment:
You shouldn't be using globals anymore when moving this to a module. Just hand it in as a parameter like in the other functions although I admit that this is a bit of a pain. Another option would be to move the whole code in this file in a class and hand in $aCmdResult in the constructor, then you don't have to worry any more at all handing it around.